hypothesis-python/src/hypothesis/strategies/_internal/core.py

```diff
@@ -1073,7 +1072,7 @@ def text(
         )
     if (max_size == 0 or char_strategy.is_empty) and not min_size:
         return just("")
-    return StringStrategy(lists(char_strategy, min_size=min_size, max_size=max_size))
+    return lists(char_strategy, min_size=min_size, max_size=max_size).map("".join)
```

Comment:
This change seems unrelated?

Reply:
I'll admit it's tangential, but `repr("")` is now `''` not `u''` and the rest is refactoring.

Reply:
Fair enough.